File: examples/key/sign/sign.py
from bloock.client.authenticity import AuthenticityClient
from bloock.client.record import RecordClient
from bloock.client.key import KeyClient
from bloock.entity.key.key_type import KeyType
from bloock.entity.authenticity.signer import Signer

if __name__ == "__main__":
    authenticity_client = AuthenticityClient()
    record_client = RecordClient()
    key_client = KeyClient()

    key = key_client.new_local_key(KeyType.EcP256k)

    signed_record = (
        record_client.from_string("Hello world").with_signer(Signer(key)).build()
    )

    print("Record was signed successfully")

    # we can add another signature with a different key

    key = key_client.new_local_key(KeyType.EcP256k)

    print("Adding another signature")
    signed_record = (
        record_client.from_record(signed_record).with_signer(Signer(key)).build()
    )

    print("Record was signed successfully")

    record_hash = signed_record.get_hash()

    print(f"Hash: {record_hash}")

    signatures = authenticity_client.get_signatures(signed_record)

    for i, signature in enumerate(signatures):
        print(f"Signature {i + 1}: {signature}")

File: examples/key/sign/Sign.java
import com.bloock.sdk.client.AuthenticityClient;
import com.bloock.sdk.client.KeyClient;
import com.bloock.sdk.client.RecordClient;
import com.bloock.sdk.entity.authenticity.Signer;
import com.bloock.sdk.entity.key.KeyType;
import com.bloock.sdk.entity.key.LocalKey;
import com.bloock.sdk.entity.record.Record;

public class Sign {
  public static void main(String[] args) throws Exception {
    RecordClient recordClient = new RecordClient();
    AuthenticityClient authenticityClient = new AuthenticityClient();
    KeyClient keyClient = new KeyClient();

    LocalKey localKey = keyClient.newLocalKey(KeyType.EcP256k);

    Record signedRecord = recordClient
        .fromString("Hello world")
        .withSigner(new Signer(localKey))
        .build();

    System.out.println("Record was signed successfully");

    // we can add another signature with a different key
    System.out.println("Adding another signature");
    localKey = keyClient.newLocalKey(KeyType.EcP256k);

    signedRecord = recordClient.fromRecord(signedRecord)
        .withSigner(new Signer(localKey))
        .build();

    System.out.println("Record was signed successfully");

    System.out.println("Hash: " + signedRecord.getHash());

    authenticityClient
        .getSignatures(signedRecord)
        .stream()
        .forEach(signature -> {
          System.out.println("Signature " + signature.getSignature());
        });
  }
}